app/controllers/hiring_staff/vacancies_controller.rb

```diff
@@ -32,7 +32,7 @@ def review
     unless @vacancy.valid?
       redirect_to_incomplete_step
     else
-      state = params[:edit_draft] == 'true' ? 'edit' : 'review'
+      state = params[:edit_draft] == 'true' ? 'edit' :  @vacancy&.state == 'copy' ? 'copy' : 'review'
       @vacancy.update(state: state) unless @vacancy&.state == state
```

Comment:
I think that if you try to call `update` and you set the same value, nothing happens anyway so you do not need to check that.

Reply:
Ah ok nice, thanks